### PR TITLE
adapted workflow for soilgrids2

### PR DESCRIPTION
**Pull request by laurenebouaziz**
see #21 for detailed description of changes.

### CODE REVIEW

**Comment by verseve** [comment]
I have focused on the proposed calculations for Soilgrids 2.0 and implementation is correct. One question: for the c parameter, is it perhaps better to use a weighted average here?:
https://github.com/Deltares/hydromt_wflow/blob/soilgrids2/hydromt_wflow/workflows/soilgrids.py#L435

My suggestion is to also add the calculation approaches for the different Soilgrids versions to the documentation.

**Comment by laurenebouaziz** [comment]
Willem, thanks for checking the code. not sure if I understand the question correctly, isn't the weighted average computed in lines 441-454? 
good point to document better the different calculation approaches.

**Comment by verseve** [comment]
Ah, yes you're right! I was looking at the wrong soil depth array (midpoint).

File: hydromt_wflow/workflows/soilgrids.py [hunk 1]
```diff
@@ -165,21 +155,19 @@ def pore_size_distrution_index_layers(ds, thetas):
...
-            ds["clyppt_sl" + str(l)],
-            dask="parallelized",
-            output_dtypes=[float],
-        )
-        da.name = "sl" + str(l)
-        da = da.raster.interpolate_na()
-        da_lst.append(da)
-    ds = xr.merge(da_lst)
-    return ds
+    da = xr.apply_ufunc(
```
Comment (hboisgon) [inline hydromt_wflow/workflows/soilgrids.py:158]: Use da for DataArray and ds for Dataset. To ease comprehension for the developper :)

**Review by hboisgon — COMMENTED** [review]
Nice additions to handle soilgrids 2020! 
In the end it's not that many "if soilgrids =", only during resampling over the different soil layers which is to be expected. Good that we can keep both options.
There's one comment somewhere on conventions on the use of da vs ds but minor :)

One last thing, I didn't see the yml file you used to read soilgrids data so not sure if this is relevant but there was no way to concat over soil layers direclty when reading soilgrids data (passing kwargs for the concat and concat_dim of this function https://github.com/Deltares/hydromt/blob/a3f5dc16248e4335ef89df749d87e2b72985c97c/hydromt/io.py#L70 )?

**Comment by laurenebouaziz** [comment]
thanks! 
I have adapted whenever applicable da to ds.
We had checked with Dirk, but it was not possible to right away use concat_dim to merge the layers, as far as I understood. 

Here is the yaml for just soilgrids_2020 to be added to the artifacts
[soilgrids_2020.zip](https://github.com/Deltares/hydromt_wflow/files/6632023/soilgrids_2020.zip)

**Comment by laurenebouaziz** [comment]
If I see it correctly, tests are failing on the wflow_soil.map as this map was completely changed.
@hboisgon , do we need to change the tests or should we change wflow_soil to soil_texture  and keep the previous map?

**Comment by hboisgon** [comment]
The tests are here to check that when changing the code you do not get unexpected changes or results downstream. 

Here you changed the computation method for wflow_soil.map so that the changes for this map is expected :) as this is something we want to keep from now on, then you need to update the examples used for the tests as well (here this means update the staticmaps of all models: wflow_piave_subbasin, wflow_piave_clip and wflow_piave_sediment).

**Pull request merged**